linera-client/src/client_context.rs

```diff
@@ -317,24 +332,24 @@ where
     pub async fn update_wallet_for_new_chain(
         &mut self,
         chain_id: ChainId,
-        key_pair: Option<AccountSecretKey>,
+        owner: Option<AccountOwner>,
```

Comment:
I think this should now be just non-optional `AccountOwner` (and down the callstack).  I don't think it makes sense to have a chain without an owner.

Reply:
And non-user chains might have a more complicated ownership structure than just one `AccountOwner`, right.